src/main/java/org/alliancegenome/curation_api/services/base/BaseOntologyTermService.java

```diff
@@ -201,30 +211,39 @@ private void handleCrossReferences(OntologyTerm dbTerm, OntologyTerm incomingTer
 	}
 	
 	private void handleSynonyms(OntologyTerm dbTerm, OntologyTerm incomingTerm) {
-		Set<String> currentSynonyms;
+		Set<Synonym> currentSynonyms;
 		if(dbTerm.getSynonyms() == null) {
 			currentSynonyms = new HashSet<>();
 			dbTerm.setSynonyms(new ArrayList<>());
 		} else {
 			currentSynonyms = dbTerm.getSynonyms().stream().collect(Collectors.toSet());
 		}
+		List<String> currentSynonymNames = currentSynonyms.stream().map(Synonym::getName).collect(Collectors.toList());
 		
-		Set<String> newSynonyms;
+		Set<Synonym> newSynonyms;
 		if(incomingTerm.getSynonyms() == null) {
 			newSynonyms = new HashSet<>();
 		} else {
 			newSynonyms = incomingTerm.getSynonyms().stream().collect(Collectors.toSet());
 		}
+		List<String> newSynonymNames = currentSynonyms.stream().map(Synonym::getName).collect(Collectors.toList());
```

Comment:
Should have caught this:

`List<String> newSynonymNames = currentSynonyms.stream().map(Synonym::getName).collect(Collectors.toList());`

should be:

`List<String> newSynonymNames = newSynonyms.stream().map(Synonym::getName).collect(Collectors.toList());`

Reply:
Indeed!